components/proxy/src/test/scala/com/hotels/styx/ServerConfigSchemaSpec.scala

```diff
@@ -0,0 +1,360 @@
+      validateServerConfiguration(yamlConfig(
+        minimalConfig
+          +
+          """
+            |loadBalancing:
+            |  x: false
+          """.stripMargin
+      )) should be(Optional.empty())
+    }
+
+    it("Accepts 'url' field - as an OBJECT") {
+      validateServerConfiguration(yamlConfig(
+        minimalConfig
+          +
+          """
+            |url:
+            |  encoding:
+            |    unwiseCharactersToEncode: '43'
+          """.stripMargin
+      )) should be(Optional.empty())
+    }
+
+    it("Accepts 'originRestrictionCookie' field - as a STRING") {
+      validateServerConfiguration(yamlConfig(
+        minimalConfig
+          +
+          """
+            |originRestrictionCookie: mycookie
+          """.stripMargin
+      )) should be(Optional.empty())
+    }
+
+    it("Accepts 'plugins' field - but does not validate") {
+      validateServerConfiguration(yamlConfig(
+        minimalConfig
+          +
+          """
+            |plugins:
+            |  active: plugA, plugB
+            |  all:
+            |    plugA:
+            |      x
+            |    plugB:
+            |      y
+          """.stripMargin
+      )) should be(Optional.empty())
+    }
+
+    it("Accepts 'logFormat' field as a STRING") {
+      validateServerConfiguration(yamlConfig(
+        minimalConfig
+          +
+          """
+            |plugins:
+            |  active: plugA, plugB
+            |  all:
+            |    plugA:
+            |      x
+            |    plugB:
+            |      y
+          """.stripMargin
+      )) should be(Optional.empty())
+    }
+
+    it("Accepts 'styxHeaders' field as an OBJECT") {
+      validateServerConfiguration(yamlConfig(
+        minimalConfig
+          +
+          """
+            |styxHeaders:
+            |  styxInfo:
+            |    name: "X-Styx-Info"
+            |    format: "xyz"
+            |  originId:
+            |    name: "X-Origin-Id"
+            |  requestId:
+            |    name: "X-Request-Id"
+          """.stripMargin
+      )) should be(Optional.empty())
+    }
+
+  }
+
+  describe("StyxHeaders Object") {
+    it("Aaccepts 'format' field for 'styxInfo' only") {
+      validateServerConfiguration(yamlConfig(
+        minimalConfig
+          +
+          """
+            |styxHeaders:
+            |  originId:
+            |    name: "X-Origin-Id"
+            |    format: "x"
+          """.stripMargin
+      )) should be(Optional.of("Unexpected field: 'styxHeaders.originId.format'"))
+
+      validateServerConfiguration(yamlConfig(
+        minimalConfig
+          +
+          """
+            |styxHeaders:
+            |  requestId:
+            |    name: "X-Request-Id"
+            |    format: "x"
+          """.stripMargin
+      )) should be(Optional.of("Unexpected field: 'styxHeaders.requestId.format'"))
+    }
+
+    it("'format' field is optional for 'styxInfo'") {
```

Comment:
The word `it` plus the string argument should form a sentence.

Reply:
Addressed this.